packages/ra-core/src/form/FormDataConsumer.tsx

```diff
@@ -43,22 +43,25 @@ import warning from '../util/warning';
  * );
  */
 const FormDataConsumer = (props: ConnectedProps) => {
-    const formData = useWatch();
+    const { getValues } = useFormContext();
+    let formData = useWatch();
```

Comment:
`useWatch` docs: `Note: the first render will always return defaultValue when it's supplied.`

Initial discussion:
https://github.com/react-hook-form/react-hook-form/issues/3758

So, is not a bug, is a feature of react-hook-form.